### DISCUSSION TITLE
Causes Preferences.h puts to fail on ESP32 Dev and ESP32S boards

### DISCUSSION THREAD

**Discussion by plrfishing** [taligentx/dscKeybusInterface#273] [dscKeybusInterface Discussion]
Have been trouble shooting Preferences.h while using dsckeybusinterface. Experience failure on put values when ESP32 is connected to dsc PC-1616 system. Puts do not fail if dsc connections removed or if dsckeybusinterface.h is not part of code.

**Comment by plrfishing** [comment]
Ok, use dsc.stop(); before preferences put value and dsc.begin(); afterwards seems to allow puts to work properly.

**Comment by taligentx** [comment]
Hi @plrfishing - does the esp32 crash during the put?  The interrupt functions used by dscKeybusInterface should be in IRAM but I'm curious if something else is being called while the esp32 is accessing flash.  Stopping and restarting the interface should work fine if there's no important activity that needs to be captured on the Keybus in the interval.

**Reply by plrfishing** [reply]
I figured out the problem. It was caused by the ESP32 that I used had a silkscreen label of GND for a pin that is actually CMD. When I attached the ground wire from the DSC system to the actual ESP32 ground all works perfectly.

**Reply by taligentx** [reply]
Great to hear it's resolved, and congrats on figuring out such a misleading GND label!  I imagine that esp32 board is going to cause a lot of headaches, it'd be tough to catch by remote troubleshooting.
